### PR TITLE
feat(animal): add petName method

### LINKED ISSUE #2206: Pet names
### Clear and concise description of the problem

There aren't pet names

### Suggested solution

Add pet names

### Alternative

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by suyashgulati**
To resolve #2206

Adds new method `petName` on animal module

### CODE REVIEW

**Comment by suyashgulati** [comment]
Need maintainers' help on resolving the test failure:

```
Error: The locale data for 'animal.petName' are missing in this locale.
  Please contribute the missing data to the project or use a locale/Faker instance that has these data.
  For more information see https://fakerjs.dev/guide/localization.html
```
  
I am not sure what needs to be done for this. Do I need to add any other locale than en to resolve this?

File: src/modules/animal/index.ts [hunk 1]
```diff
@@ -72,6 +72,20 @@ export class AnimalModule extends ModuleBase {
     return this.faker.helpers.arrayElement(this.faker.definitions.animal.lion);
   }
 
+  /**
+   * Returns a random pet name.
+   *
+   * @example
+   * faker.animal.petName() // 'Coco'
+   *
+   * @since 5.5.0
```
Comment (matthewmayer) [inline src/modules/animal/index.ts]: Should be `@since 9.2.0` as that will be next minor release

File: test/modules/animal.spec.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ const functionNames = [
   'horse',
   'insect',
   'lion',
+  'petName',
```
Comment (matthewmayer) [inline test/modules/animal.spec.ts]: the problem with the test failure is that these tests assume that the definition name exactly matches the method name. But in this case it (correctly) doesn't - petName vs pet_name. I would remove petName from the functionNames array and write a seperate test for it.

**Review by matthewmayer — CHANGES_REQUESTED** [review]

File: test/modules/animal.spec.ts (hunk 2 above)
Comment (ST-DDT) [reply test/modules/animal.spec.ts]: I'll have a look at it.

**Comment by ST-DDT** [comment]
Blocked by: #3197

- #3197

File: test/modules/animal.spec.ts (hunk 2 above)
Comment (ST-DDT) [reply test/modules/animal.spec.ts]: -> #3197

**Comment by ST-DDT** [comment]
This is no longer blocked, but you have to adjust the test file.

File: test/modules/animal.spec.ts [hunk 3]
```diff
@@ -133,6 +134,13 @@ describe('animal', () => {
           expect(faker.definitions.animal.type).toContain(actual);
         });
       });
+
+      describe('petName()', () => {
+        it('should return random value from type array', () => {
```
Comment (matthewmayer) [inline test/modules/animal.spec.ts]: From pet name array not type array?

File: src/locales/en/animal/pet_name.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+export default [
+  'Ace',
+  'Archie',
+  'Bailey',
+  'Bandit',
+  'Bear',
```
Comment (matthewmayer) [inline src/locales/en/animal/pet_name.ts]: Maybe omit this just because it's confusing if this is the value (You might think you have the wrong method?)

**Review by matthewmayer — CHANGES_REQUESTED** [review]

File: test/modules/animal.spec.ts (hunk 3 above)
Comment (ST-DDT) [reply test/modules/animal.spec.ts]:
```suggestion
        it('should return random value from pet name array', () => {
```

File: src/locales/en/animal/pet_name.ts (hunk 4 above)
Comment (ST-DDT) [reply src/locales/en/animal/pet_name.ts]:
```suggestion
```

**Review by ST-DDT — APPROVED** [review]

**Review by matthewmayer — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Comment by ST-DDT** [comment]
Thanks for your contribution. ❤️ 
Your additions will be available in Faker v9.2.0.
While I don't have an exact release date for it yet, we will likely release it somewhat soonish (a week - a few weeks).

**Pull request merged**